doc/user/content/sql/create-source/mysql.md

```diff
@@ -0,0 +1,387 @@
+replicating `schema1.table_1` and `schema2.table_1`. Use the `FOR TABLES`
+clause to provide aliases for each upstream table, in such cases, or to specify
+an alternative destination schema in Materialize.
+
+```sql
+CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_connection
+  FOR TABLES (schema1.table_1 AS s1_table_1, schema2_table_1 AS s2_table_1);
+```
+
+### Monitoring source progress
+
+[//]: # "TODO(morsapaes) Replace this section with guidance using the new
+progress metrics in mz_source_statistics + console monitoring, when available
+(also for PostgreSQL)."
+
+By default, MySQL sources expose progress metadata as a subsource that you
+can use to monitor source **ingestion progress**. The name of the progress
+subsource can be specified when creating a source using the `EXPOSE PROGRESS
+AS` clause; otherwise, it will be named `<src_name>_progress`.
+
+The following metadata is available for each source as a progress subsource:
+
+Field              | Type                                                    | Details
+-------------------|---------------------------------------------------------|--------------
+`source_id_lower`  | [`uuid`](https://materialize.com/docs/sql/types/uuid/)  |
+`source_id_upper`  | [`uuid`](https://materialize.com/docs/sql/types/uuid/)  |
+`transaction_id`   | [`uint8`](/sql/types/uint/#uint8-info)
+
+And can be queried using:
+
+```sql
+SELECT transaction_id
+FROM <src_name>_progress;
+```
+
+The reported transaction ID should increase as Materialize consumes **new**
+binlog records from the upstream MySQL database. For more details on
+monitoring source ingestion progress and debugging related issues, see
+[Troubleshooting](/ops/troubleshooting/).
+
+## Known limitations
+
+##### Schema changes
+
+{{% mysql-direct/mysql-schema-changes %}}
+
+##### Supported types
+
+Materialize natively supports the following MySQL types:
+
+<ul style="column-count: 3">
+<li><code>bigint</code></li>
+<li><code>binary</code></li>
+<li><code>blob</code></li>
+<li><code>boolean</code></li>
+<li><code>char</code></li>
+<li><code>date</code></li>
+<li><code>datetime</code></li>
+<li><code>decimal</code></li>
+<li><code>double</code></li>
+<li><code>float</code></li>
+<li><code>int</code></li>
+<li><code>json</code></li>
+<li><code>longblob</code></li>
+<li><code>longtext</code></li>
+<li><code>mediumblob</code></li>
+<li><code>mediumint</code></li>
+<li><code>mediumtext</code></li>
+<li><code>numeric</code></li>
+<li><code>real</code></li>
+<li><code>smallint</code></li>
+<li><code>text</code></li>
+<li><code>time</code></li>
+<li><code>timestamp</code></li>
+<li><code>tinyblob</code></li>
+<li><code>tinyint</code></li>
+<li><code>tinytext</code></li>
+<li><code>varbinary</code></li>
+<li><code>varchar</code></li>
+</ul>
+
+Replicating tables that contain **unsupported [data types](/sql/types/)** is
+possible via the `TEXT COLUMNS` option. The specified columns will be treated
```

Comment:
Also, I think here we'll want to explicitly call out the exact data types we support via `TEXT COLUMNS`, because AFAIU we have to intentionally decode each unsupported type in the code (unlike for PostgreSQL), right? @rjobanp

Reply:
Agreed -- we should explicitly state that right now we support json, enum, and year types as text